pkg/serializer/frame/tracing.go

```diff
@@ -0,0 +1,74 @@
+package frame
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/weaveworks/libgitops/pkg/tracing"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	// Constants shown in e.g. the Jaeger UI
+	pkgPrefix    = "frame."
+	readerPrefix = pkgPrefix + "Reader"
+	writerPrefix = pkgPrefix + "Writer"
+)
+
+func spanAssociateFrameBytes(span trace.Span, len int) {
+	span.SetAttributes(attribute.Int("frame_bytes", len))
+}
+
+func closeWithTrace(ctx context.Context, fnTracer tracing.FuncTracer, c Closer, hasCloser bool) tracing.TraceFuncResult {
+	spanName := "Close"
+	// Mark clearly that if hasCloser == false, no closing operation was performed
+	if !hasCloser {
+		spanName = "CloseNoop"
+	}
+	return fnTracer.TraceFunc(ctx, spanName, func(ctx context.Context, _ trace.Span) error {
+		// Don't close if a closer wasn't given
```

Comment:
You could probably simplify this code and remove the comments by inverting the logic.